cf/data/data.py

```diff
@@ -4407,7 +4407,9 @@ def arctan2(cls, x1, x2):
             x2: array_like
                 X coordinates. *x1* and *x2* must be broadcastable to
                 a common shape (which becomes the shape of the
-                output).
+                output). If both *x1* and *x2* have units, they must
+                be in the same dimension (can be conformed), else
+                they will be treated as unitless.
```

Comment:
A good use of dimensional as in 'dimensional analysis', but in this context (when considering multidimensional data arrays as is central to cf-python), 'dimension' already has a specific meaning, namely the [(number of) array axes](https://numpy.org/doc/stable/glossary.html#term-dimension), so best avoid using it in the first context.

We tend to use the word 'equivalent' to mean units that can be conformed i.e. converted by scaling values e.g. 'kg' and 'g' as opposed to 'kg' and 's' or 'apples' (see https://ncas-cms.github.io/cfunits/cfunits.Units.html if you want to learn more), so I suggest:

```suggestion
                be equal or equivalent, in which case they
                will be conformed.
```

(Also no need to cover what happens if not so I suggest dropping that end to the sentence.)